src/Spice86.Core/Emulator/OperatingSystem/DosFileManager.cs

```diff
@@ -717,4 +762,43 @@ public class DosFileManager {
         dosDiskTransferArea.FileSize = (ushort)attributes.Length;
         dosDiskTransferArea.FileName = Path.GetFileName(matchingFile);
     }
+
+    /// <summary>
+    /// Creates a directory on disk.
+    /// </summary>
+    /// <param name="directory">The directory name to create</param>
+    /// <returns></returns>
+    /// <returns>A <see cref="DosFileOperationResult"/> with details about the result of the operation.</returns>
+    public DosFileOperationResult CreateDirectory(string directory) {
+        string caseInsensitivePath = ToHostFilePath(directory);
+
+        string? fullPath = ToHostCaseSensitiveFileName(directory, false);
+        if (string.IsNullOrWhiteSpace(fullPath)) {
+            return FileNotFoundError(directory);
+        }
+
+        DirectoryInfo? parentFolder = Directory.GetParent(fullPath);
+
+        if (parentFolder?.Exists is false or null) {
+            return FileNotFoundError(directory);
+        }
+
+        if (parentFolder.GetDirectories().Select(x => x.Name)
+            .Concat(parentFolder.GetFiles().Select(x => x.Name))
+            .Any(x => string.Equals(x, directory, StringComparison.InvariantCultureIgnoreCase))) {
```

Comment:
directory can contain more than the directory name isnt it? In this case I am not sure to understand how it is going to be equal to files / folders names

Reply:
It lists folders first, files next, takes their names, and determines if any equals with the new directory name.